Orange/widgets/data/owconcatenate.py

```diff
@@ -152,8 +155,34 @@ def set_more_data(self, data=None, sig_id=None):
             del self.more_data[sig_id]
 
     def handleNewSignals(self):
+        self.Error.clear()
         self.mergebox.setDisabled(self.primary_data is not None)
-        self.unconditional_apply()
+        incompatible = self.incompatible_types()
+        if incompatible:
```

Comment:
Probably just `if self.incompatible_types():` ?

Reply:
Yes, thank you, i refactored code and then missed this.